lib/model/autocomplete.dart

```diff
@@ -246,7 +246,23 @@ class MentionAutocompleteView extends ChangeNotifier {
 
   Future<List<MentionAutocompleteResult>?> _computeResults(MentionAutocompleteQuery query) async {
     final List<MentionAutocompleteResult> results = [];
-    final Iterable<User> users = store.users.values;
+    final Set<int> userIds;
+
+    switch (narrow) {
+      case AllMessagesNarrow():
+        userIds = {};
+      case StreamNarrow():
+        userIds = store.dmUserIds;
+        userIds.addAll(store.streamUserIds[(narrow as StreamNarrow).streamId] ?? {});
+      case TopicNarrow():
+        userIds = store.dmUserIds;
+        userIds.addAll(store.streamUserIds[(narrow as TopicNarrow).streamId] ?? {});
```

Comment:
Have you looked into the web client's logic for this? I believe its top priority when composing to a topic narrow is the participants in that individual topic, rather than the whole stream.